src/Helper/Service/IdentityFactory.php

```diff
@@ -25,10 +26,22 @@ class IdentityFactory implements FactoryInterface
      */
     public function __invoke(ContainerInterface $container, $name, array $options = null)
     {
+        $container = $container->getServiceLocator();
```

Comment:
This will throw an `E_USER_DEPRECATED` in v3. Is it really needed though? AFAIK in v3 this should be called with the `creationContext` of the container - i.e. the "root" ServiceManager if called from a plugin manager.

But I'm guessing you added it for a reason. If you need to distinguish between v2 and v3 the only real solution I've come up with is duck-typing the `configure()` method (which is only in v3). Ugly, but it works...

```
if (! method_exists($container, 'configure')) {
    $container = $container->getServiceLocator();
}
```